tensorflow_io/core/kernels/audio_flac_kernels.cc

```diff
@@ -262,6 +310,180 @@ class FlacReadableResource : public AudioReadableResourceBase {
+
+    int32 rate;
+    DataType dtype;
+    TensorShape shape;
+    OP_REQUIRES_OK(context, resource->Spec(&shape, &dtype, &rate));
+
+    OP_REQUIRES(context, (dtype == context->expected_output_dtype(0)),
+                errors::InvalidArgument(
+                    "dtype mismatch: ", DataTypeString(dtype), " vs. ",
+                    DataTypeString(context->expected_output_dtype(0))));
+
+    PartialTensorShape provided_shape;
+    OP_REQUIRES_OK(context, PartialTensorShape::MakePartialShape(
+                                shape_tensor->flat<int64>().data(),
+                                shape_tensor->NumElements(), &provided_shape));
+    OP_REQUIRES(context, (provided_shape.IsCompatibleWith(shape)),
+                errors::InvalidArgument(
+                    "shape mismatch: ", provided_shape.DebugString(), " vs. ",
+                    shape.DebugString()));
+
+    OP_REQUIRES_OK(
+        context,
+        resource->Read(0, shape.dim_size(0),
+                       [&](const TensorShape& shape, Tensor** value) -> Status {
+                         TF_RETURN_IF_ERROR(
+                             context->allocate_output(0, shape, value));
+                         return Status::OK();
+                       }));
+  }
+
+ private:
+  mutable mutex mu_;
+  Env* env_ GUARDED_BY(mu_);
+};
+class AudioEncodeFlacOp : public OpKernel {
+ public:
+  explicit AudioEncodeFlacOp(OpKernelConstruction* context)
+      : OpKernel(context) {
+    env_ = context->env();
+  }
+
+  void Compute(OpKernelContext* context) override {
+    const Tensor* input_tensor;
+    OP_REQUIRES_OK(context, context->input("input", &input_tensor));
+
+    const Tensor* rate_tensor;
+    OP_REQUIRES_OK(context, context->input("rate", &rate_tensor));
+
+    const int64 rate = rate_tensor->scalar<int64>()();
+    const int64 samples = input_tensor->shape().dim_size(0);
+    const int64 channels = input_tensor->shape().dim_size(1);
+
+    const int64 bytes_per_sample = DataTypeSize(input_tensor->dtype());
+    // TODO: support more data types
+    switch (input_tensor->dtype()) {
+      case DT_INT16:
+        break;
+      default:
+        OP_REQUIRES(context, false,
+                    errors::InvalidArgument(
+                        "data type ", DataTypeString(input_tensor->dtype()),
+                        " not supported"));
+    }
+
+    std::unique_ptr<FLAC__StreamEncoder, void (*)(FLAC__StreamEncoder*)>
+        encoder(nullptr, [](FLAC__StreamEncoder* p) {
+          if (p != nullptr) {
+            FLAC__stream_encoder_delete(p);
+          }
+        });
+    encoder.reset(FLAC__stream_encoder_new());
+
+    FLAC__bool ok;
+
+    ok = FLAC__stream_encoder_set_verify(encoder.get(), true);
+    OP_REQUIRES(context, ok, errors::InvalidArgument("unable to set verify"));
+
+    // ok = FLAC__stream_encoder_set_compression_level(encoder.get(), 5);
+    // OP_REQUIRES(context, ok, errors::InvalidArgument("unable to set
+    // compression level"));
```

Comment:
Are these commented code needed?

Reply:
@terrytangyuan I am thinking the compression level could be an input value passed from user. Not sure if it makes sense yet, so comment out.

I have added a TODO comment in the code.